Update SDK & API

PrXenon opened the pull request:
Update to latest  Android SDK und API
minSDK now 23
targetSDK 33

add versionnumber to plugins
chaged the location for external plugins (testing)

PrXenon committed:
PrXenon Updates

PrXenon committed:
PrXenon Updates

PrXenon committed:
PrXenon Updates

PrXenon committed:
PrXenon Updates

modos189 commented on mobile/app/build.gradle:
```diff
@@ -24,15 +24,14 @@ def metaFDroid(outputFileName) {
+        targetSdkVersion 33
         versionCode = getVersionCodeTimeStamps()
-        versionName "0.34.1"
+        versionName "0.40.1"
```
Please remove the version change, since a new number will be assigned when a new version of the IITC is released

PrXenon replied:
Ok.. i understand.. will do it

modos189 commented on mobile/app/build.gradle:
```diff
@@ -60,7 +59,7 @@ android {
         debug {
             applicationIdSuffix ".debug"
-            versionNameSuffix "-test-$BUILD_DATE"
+            versionNameSuffix "-test-px-$BUILD_DATE"
```
Apparently this is an unnecessary change

PrXenon replied:
Is a relict so i can install official version and this test side by side.. 

I change it back

modos189 commented on mobile/app/build.gradle:
```diff
@@ -78,33 +77,35 @@ android {
         variant.getAssembleProvider().configure() {
-            it.doLast {
-                metaFDroid(variant.getPackageApplicationProvider().get().outputDirectory)
+            def outputDir = new File("${project.buildDir.absolutePath}/outputs/apk/${variant.flavorName}/${variant.buildType.name}")
```
Won't this cause an error when building an aab instead of an apk?

PrXenon replied:
I have to test it

modos189 commented on mobile/app/src/debug/res/values/app_name.xml:
```diff
@@ -1,4 +1,4 @@
 <?xml version="1.0" encoding="utf-8"?>
 <resources>
-    <string name="app_name">IITC Mobile Test</string>
+    <string name="app_name">IITC Mobile</string>
```
This change is not necessary because the debug version should have the "Test" postfix

PrXenon replied:
Is a relict so i can install official version and this test side by side

modos189 replied:
I don't understand what the question is. The release version is "IITC Mobile", the beta version is "IITC Mobile Beta", the builds from PR or in debug mode are "IITC Mobile Test". These are three different versions that you can have installed at the same time.
In any case, you can choose any name to build on your own. But please remove this line in PR

modos189 commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java:
```diff
@@ -483,7 +508,7 @@ protected void onStart() {
                 // enough idle...let's do some work
                 Log.d("resuming...reset idleTimer");
-                mIitcWebView.loadJS("(function(){if(window.idleReset) window.idleReset();})();");
+                mIitcWebView.loadJS("(function(){if(3,,window.idleReset) window.idleReset();})();");
```
Is that a typo?

PrXenon replied:
Oh.. my bad..

modos189 commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_Mobile.java:
```diff
@@ -1253,6 +1278,16 @@ public void onRequestPermissionsResult(int requestCode, @NonNull String[] permis
                 mUserLocation.onRuntimePermissionsGranted();
             }
         }
+        if (requestCode == 355) {
```
Maybe this code should be deleted?

PrXenon replied:
Maybe.. i will add an TODO
I have trying to add the file permission but there changed so many things with sdk 33

modos189 commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_TileManager.java:
```diff
@@ -33,7 +33,7 @@ public WebResourceResponse getTile(String url) throws Exception {
         /*
          * now get the file path...example path:
-         * /storage/emulated/0/Android/data/org.exarhteam.iitc_mobile/files/mqcdn.com/tiles/1.0.0/map/18/137397/89580.jpg
+         * /storage/emulated/0/Android/data/org.exarhteam.iitc_mobile_px/files/mqcdn.com/tiles/1.0.0/map/18/137397/89580.jpg
```
No need to change the string

PrXenon replied:
Is a relict so i can install official version and this test side by side

modos189 commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebView.java:
```diff
@@ -56,9 +56,9 @@ private void iitc_init(final Context c) {
         mSettings.setAllowFileAccess(true);
         mSettings.setGeolocationEnabled(true);
-        mSettings.setAppCacheEnabled(true);
+      //  mSettings.setAppCacheEnabled(true);
```
Is this an intentional change?

PrXenon replied:
No maybe i tryed something... dont remember

modos189 commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/IITC_WebViewClient.java:
```diff
@@ -259,8 +259,12 @@ public WebResourceResponse shouldInterceptRequest(final WebView view, final Stri
+
+
             return mIitc.getFileManager().getResponse(uri);
+        Log.i("File --> + " +uri);
```
The line that's not needed

PrXenon replied:
For testing i need it.. because i cannot write to the old path to store the external user plugins.. 

Will also add a TODO here for removing the LOG

modos189 replied:
You can keep it, but delete it for PR

modos189 commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java:
```diff
@@ -140,11 +141,16 @@ public boolean onOptionsItemSelected(final MenuItem item) {
                 if (mFileManager.checkWriteStoragePermissionGranted()) {
                     // create the chooser Intent
+
+                    Log.i("ABLAGE2--> "+ Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOCUMENTS) + "/IITC_Mobile/");
```
The line that's not needed

PrXenon replied:
For testing i need it.. because i cannot write to the old path to store the external user plugins..

Will also add a TODO here for removing the LOG

modos189 commented on mobile/app/src/main/java/org/exarhteam/iitc_mobile/prefs/PluginPreferenceActivity.java:
```diff
@@ -343,7 +350,7 @@ private void addHeader(final String title, final boolean userPlugin) {
         newHeader.title = title;
         newHeader.fragmentArguments = bundle;
-        newHeader.fragment = "org.exarhteam.iitc_mobile.fragments.PluginsFragment";
+        newHeader.fragment = "org.exarhteam.iitc_mobile_px.fragments.PluginsFragment";
```
I already wrote above that it is not required

PrXenon replied:
Yep will change it back

modos189 reviewed (changes requested):
Thank you for your work. I have prepared a commentary with corrections

PrXenon committed:
requested changes

PrXenon committed:
removed old API if statements

PrXenon committed:
change back to old package name

PrXenon committed:
typo-fix

modos189 commented:
Please do not delete `mobile/keystore.properties.example`
This file is not used, but is an example for creating `mobile/keystore.properties`

PrXenon committed:
appname changes and add non translatable

PrXenon committed:
is back

PrXenon committed:
changes

PrXenon committed:
fix building
works with apk & bundle

modos189 commented:
In the new version of Gradle, when building bundle, the line `app = exec_gradle(source, dist_ext)` in build_mobile.py returns path `mobile/app/build/outputs/bundle/debug/app.aab`, as before. But the file `mobile/app/build/outputs/bundle/debug/app-debug.aab` is actually created. The specified file does not exist, so the build crashes with an error. Any idea how to fix this in gradle?

I checked that if I replace `app.aab` with `app-debug.aab` in the build script, the bundle build completes successfully.

modos189 commented:
Ah, I see. Changes in the build script are required. In order for me not to create PR, please make these changes in `build_mobile.py`:

![image](https://user-images.githubusercontent.com/1785196/211804151-88a6173e-eaaf-4597-b48d-e95da25a7795.png)

PrXenon committed:
Update build_mobile.py

added  
"build_file_name = f'app--{buildtype}.{dist_ext}' " in line 43

PrXenon committed:
Update build_mobile.py

removed old  f'app--{buildtype}.{dist_ext}' in line 49 and 53
sorry.. didnt see this

PrXenon committed:
Update build_mobile.py

typo

PrXenon committed:
app name changed back

PrXenon committed:
Update app_name.xml

changed app name back to "IITC Mobile" in PR

PrXenon committed:
Merge branch 'master' of https://github.com/PrXenon/ingress-intel-total-conversion

PrXenon committed:
with TargetSDK the File permission changed
this part is for debug and find a new way to store external user scripts

PrXenon committed:
User plugins location

PrXenon committed:
User plugins location

PrXenon committed:
User plugins location

PrXenon committed:
User plugins location
option to delete a user plugin

PrXenon committed:
ext. plugins bugfix

PrXenon committed:
Optimize Imports
Reformat Code

PrXenon committed:
Import userscript and option to delete - bugfix

PrXenon committed:
removed old skd stuff

PrXenon committed:
cleanup
